raiden/tests/utils/smoketest.py

```diff
@@ -250,67 +247,44 @@ def setup_testchain(
     p2p_port = next(free_port_generator)
     base_datadir = os.environ["RST_DATADIR"]
 
-    description = EthNodeDescription(
-        private_key=TEST_PRIVKEY,
-        rpc_port=rpc_port,
-        p2p_port=p2p_port,
-        miner=True,
-        extra_config={},
-        blockchain_type=eth_client.value,
-    )
-
     eth_rpc_endpoint = f"http://127.0.0.1:{rpc_port}"
     web3 = Web3(HTTPProvider(endpoint_uri=eth_rpc_endpoint))
     web3.middleware_stack.inject(geth_poa_middleware, layer=0)
 
-    config = eth_node_config(description.private_key, description.p2p_port, description.rpc_port)
-
-    config.update({"unlock": 0, "mine": True, "password": os.path.join(base_datadir, "pw")})
-
-    nodes_configuration = [config]
-    logdir = os.path.join(base_datadir, "logs")
-
-    # the marker is hardcoded in the genesis file
-    random_marker = remove_0x_prefix(encode_hex(b"raiden"))
-    seal_account = privatekey_to_address(description.private_key)
-    accounts_to_fund = [TEST_ACCOUNT_ADDRESS, TEST_PARTNER_ADDRESS]
+    eth_nodes = [
+        EthNodeDescription(
+            private_key=TEST_PRIVKEY,
+            rpc_port=rpc_port,
+            p2p_port=p2p_port,
+            miner=True,
+            extra_config={},
+            blockchain_type=eth_client.value,
+        )
+    ]
 
+    random_marker = remove_0x_prefix(hex(random.getrandbits(100)))
     genesis_description = GenesisDescription(
-        prefunded_accounts=accounts_to_fund,
+        prefunded_accounts=[TEST_ACCOUNT_ADDRESS, TEST_PARTNER_ADDRESS],
         random_marker=random_marker,
         chain_id=NETWORKNAME_TO_ID["smoketest"],
     )
 
+    nodekeyhex = remove_0x_prefix(encode_hex(TEST_PRIVKEY))
+    datadir = eth_node_to_datadir(nodekeyhex, base_datadir)
     if eth_client is EthClient.GETH:
```

Comment:
This is done by `run_private_blockchain`. I couldn't remove the special code paths without many changes, they are needed to start the raiden smoketest node.